#### Problem Statement 1:
Is gender independent of education level? A random sample of 395 people were
surveyed and each person was asked to report the highest education level they
obtained. The data that resulted from the survey is summarized in the following table:

High School   Bachelors   Masters   Ph.d.   Total

Female 60 54 46 41 201

Male 40 44 53 57 194

Total 100 98 99 98 395

Question: Are gender and education level dependent at 5% level of significance? In
other words, given the data collected above, is there a relationship between the
gender of an individual and the level of education that they have obtained?

In [1]:
import scipy.stats as sts
from scipy.stats import norm
import math
import numpy as np
import pandas as pd

In [5]:
female = [60, 54, 46, 41]
male = [40, 44, 53, 57]

school= [40, 60]
bachelors = [44, 54]
masters = [57, 46]
phd = [57, 41]

marks = male + female
sex =  ['Male','Male','Male','Male','Female','Female','Female','Female']
edu = ['High School', 'Bachelors', 'Masters', 'Ph.d.','High School', 'Bachelors', 'Masters', 'Ph.d.']
df_edu = pd.DataFrame({"Sex":sex,"Edu":edu,"Marks":marks})

df_edu

Sex          Edu  Marks
0    Male  High School     40
1    Male    Bachelors     44
2    Male      Masters     53
3    Male        Ph.d.     57
4  Female  High School     60
5  Female    Bachelors     54
6  Female      Masters     46
7  Female        Ph.d.     41

In [6]:
df2 = pd.crosstab(df_edu.Sex, df_edu.Edu,df_edu.Marks, aggfunc="sum",margins=True)

df2.columns = ["Bachelors","High School","Masters","Ph.d.","row_totals"]

df2.index = ["Female","Male","col_totals"]

df2

Bachelors  High School  Masters  Ph.d.  row_totals
Female             54           60       46     41         201
Male               44           40       53     57         194
col_totals         98          100       99     98         395

In [7]:
observed = df2.iloc[0:2,0:4]
observed

Bachelors  High School  Masters  Ph.d.
Female         54           60       46     41
Male           44           40       53     57

In [8]:
expected =  np.outer(df2["row_totals"][0:2],
                     df2.loc["col_totals"][0:4]) / 395.0
expected = pd.DataFrame(expected)
expected.columns = ["Bachelors","High School","Masters","Ph.d."]
expected.index = ["Female","Male"]
expected

Bachelors  High School    Masters      Ph.d.
Female  49.868354    50.886076  50.377215  49.868354
Male    48.131646    49.113924  48.622785  48.131646

In [9]:
chi_squared_stat = (((observed-expected)**2)/expected).sum().sum()

print(chi_squared_stat)

8.006066246262538


In [10]:
crit = sts.chi2.ppf(q = 0.95, df = 3)   

print("Critical value")
print(crit)

p_value = 1 - sts.chi2.cdf(x=chi_squared_stat, df=3)  
                            
print("P value")
print(p_value)

Critical value
7.814727903251179
P value
0.04588650089174717


In [11]:
sts.chi2_contingency(observed= observed)

(8.006066246262538,
 0.045886500891747214,
 3,
 array([[49.86835443, 50.88607595, 50.37721519, 49.86835443],
        [48.13164557, 49.11392405, 48.62278481, 48.13164557]]))

Because 8.006 > 7.815 we can reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.

#### Problem Statement 2:
Using the following data, perform a oneway analysis of variance using α=.05. Write
up the results in APA format.

[Group1: 51, 45, 33, 45, 67]

[Group2: 23, 43, 23, 43, 45]

[Group3: 56, 76, 74, 87, 56]

In [14]:
Group1= [51, 45, 33, 45, 67]

Group2= [23, 43, 23, 43, 45]

Group3= [56, 76, 74, 87, 56]

In [16]:
statistic, pvalue = sts.f_oneway(Group1,Group2,Group3)
print("F Statistic value {} , p-value {}".format(statistic,pvalue))

F Statistic value 9.747205503009463 , p-value 0.0030597541434430556


P value is less than 0.05. The test result suggests the groups don't have the same sample means in this case, since the p-value is significant at a 99% confidence level.

#### Problem Statement 3:
Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.
For 10, 20, 30, 40, 50:

In [17]:
sts.f_oneway([10, 20, 30, 40, 50],[5,10,15, 20, 25])

F_onewayResult(statistic=3.6, pvalue=0.0943497728424377)

In [19]:
Group1 = [10, 20, 30, 40, 50]
Group2 = [5,10,15, 20, 25]

mean_1 = np.mean(Group1)
mean_2 = np.mean(Group2)

grp1_sub_mean1 = []
grp2_sub_mean2 = []
add1 = 0
add2 = 0
for items in Group1:
    add1 += (items - mean_1)**2
for items in Group2:
    add2 += (items - mean_2)**2
var1 = add1/(len(Group1)-1)
var2 = add2/(len(Group2)-1)

F_Test = var1/var2
print("F Test is: {}".format(F_Test))

F Test is: 4.0
